### DISCUSSION TITLE
Linux Mint version numbering scheme change

### DISCUSSION THREAD

**Discussion by leonsk29** [linuxmint/discussions#540] [Ideas]
I think the Linux Mint version numbering scheme should be changed in the future to better reflect the exact date when a particular release has been launched, as this gives users more information just by looking the version number. Other Linux distros have already done this and I think Mint should follow suit, IMO, so I propose the following change:

The version numbering scheme should be changed from this:

Linux Mint 22 Cinnamon/MATE/Xfce,

to:

Linux Mint **2024.7** Cinnamon/MATE/Xfce, reflecting that this is Linux Mint, **released on July 2024**, so you know it's recent depending on the current date.

There, that's the suggestion. Devs, please take it into consideration. Thanks in advance.

**Comment by awvalenti** [comment]
Same thought here. Much simpler, less brainwork.

I'd just suggest only 25 for 2025. That would make it trivial to know both the year of Mint release and which Ubuntu version it's derived from.

Lots of apps that run on Mint are actually targeted at Ubuntu. Example: VirtualBox. This page https://www.virtualbox.org/wiki/Linux_Downloads offers these options: Ubuntu 25.04, ​Ubuntu 24.04, ​Ubuntu 22.04. How do we know which of these apply to our current Mint version?

**Reply by cplittlfield** [reply]
Well, if it's 2025, then the current Linux Mint is based on Ubuntu 24.04.  When the next major release of Linux Mint comes out next year, it will be based on Ubuntu 26.04, which is the next LTS Ubuntu version.

If you really need to be sure of the Ubuntu base version of your Linux Mint installation, run `inxi -Sxxx`

**Reply by awvalenti** [reply]
Correct. The new versioning scheme would possibly start in 2026, so Mint would jump from 22 to 26.

Tried the command and it works. `inxi -Sx` also displays the info. `cat /etc/debian_version /etc/upstream-release/lsb-release /etc/lsb-release` also help. One can also use the formula `m = u / 2 + 10` to convert from Ubuntu version to Mint version.

But nothing like seeing something like `Mint 28.2` and just knowing that it was released in 2028 based on Ubuntu also released in 2028.

**Comment by mixmastamyk** [comment]
Yes, becoming an industry standard, and with good reason.

I've thought about what to do when staying on an LTS for several years.  Ideally you'd like the start year to stay the same, and the second number to increase.  So I came up with the idea of having the second number (continue to) reflect the number of months since the beginning of the start year.

But instead of carrying the extra back to the year, allow the "month digit" to increase past 12.

As you can see below, it works well for almost 8 years.  This should be fine for most releases.  But just in case someone wanted to patch a release after 8 years, they could add a third number to represent releases past 95 months.

Note the last column below, version.  This way you can stay on the "2026" branch and update to the highest minor version available.

| DATE    | M OFFSET    | Y OFFSET  | VERSION |
| ---     | :---: | :---: | :---   |
|         |    |     |           |
| 2026-04 | 00 |     |   2026.04 |
| 2026-08 | 04 |     |   2026.08 |
| 2026-12 | 08 |     |   2026.12 |
|         |    |     |           |
| 2027-04 | 12 |  1  |   2026.16 |
| 2027-08 | 16 |     |   2026.20 |
| 2027-12 | 20 |     |   2026.24 |
|         |    |     |           |
| 2028-04 | 24 |  2  |   2026.28 |
| 2028-08 | 28 |     |   2026.32 |
| 2028-12 | 32 |     |   2026.36 |
|         |    |     |           |
| 2029-04 | 36 |  3  |   2026.40 |
| 2029-08 | 40 |     |   2026.44 |
| 2029-12 | 44 |     |   2026.48 |
|         |    |     |           |
| 2030-04 | 48 |  4  |   2026.52 |
| 2030-08 | 52 |     |   2026.56 |
| 2030-12 | 56 |     |   2026.60 |
|         |    |     |           |
| 2031-04 | 60 |  5  |   2026.64 |
| 2031-08 | 64 |     |   2026.68 |
| 2031-12 | 68 |     |   2026.72 |
|         |    |     |           |
| 2032-04 | 72 |  6  |   2026.76 |
| 2032-08 | 76 |     |   2026.80 |
| 2032-12 | 80 |     |   2026.84 |
|         |    |     |           |
| 2033-04 | 84 |  7  |   2026.88 |
| 2033-08 | 88 |     |   2026.92 |
| 2033-12 | 92 |     |   2026.96 |
|         |    |     |           |
| 2034-01 | 93 |     |   2026.97 |
| 2034-02 | 94 |     |   2026.98 |
| 2034-03 | 95 |     |   2026.99 |
| 2034-04 | 96 |  8  |   2026.99.01 |
| 2034-06 | 98 |     |   2026.99.02 |


(I wrote a python script to print this, with start year as a parameter.  Let me know if interested.)
